sdl/v2.go

```diff
@@ -24,11 +52,92 @@ type v2ExposeTo struct {
 }
 
 type v2Expose struct {
-	Port   uint16
-	As     uint16
-	Proto  string       `yaml:"proto,omitempty"`
-	To     []v2ExposeTo `yaml:"to,omitempty"`
-	Accept v2Accept     `yaml:"accept"`
+	Port        uint16
+	As          uint16
+	Proto       string        `yaml:"proto,omitempty"`
+	To          []v2ExposeTo  `yaml:"to,omitempty"`
+	Accept      v2Accept      `yaml:"accept"`
+	HTTPOptions v2HTTPOptions `yaml:"http_options"`
+}
+
+type v2HTTPOptions struct {
+	MaxBodySize uint32   `yaml:"max_body_size"`
+	ReadTimeout uint32   `yaml:"read_timeout"`
+	SendTimeout uint32   `yaml:"send_timeout"`
+	NextTries   uint32   `yaml:"next_tries"`
+	NextTimeout uint32   `yaml:"next_timeout"`
+	NextCases   []string `yaml:"next_cases"`
+}
+
+func (ho v2HTTPOptions) any() bool {
+	return len(ho.NextCases) != 0 ||
+		ho.MaxBodySize != 0 ||
+		ho.ReadTimeout != 0 ||
+		ho.SendTimeout != 0 ||
+		ho.NextTries != 0 ||
+		ho.NextTimeout != 0
+}
+
+func (ho v2HTTPOptions) asManifest() (manifest.ServiceExposeHTTPOptions, error) {
+	maxBodySize := ho.MaxBodySize
+	if 0 == maxBodySize {
+		maxBodySize = defaultMaxBodySize
```

Comment:
these defaults would ideally be configurable

Reply:
On chain or per provider ?

Reply:
I think per provider.

forget it, this is fine.